### ISSUE TITLE
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xf4 in position 30: invalid continuiation byte

### ISSUE THREAD

**Issue by roye10** [shap/shap#1449]
**My input:** explainer = shap.TreeExplainer(model,background_data,model_output = 'probability')
- whereby model is an xgboost model with the following parameters:

```python
params = {'validate_parameters': False, 'eta': 0.1, 'max_depth': 3, 'objective': 'binary:hinge'}
```

and the background data is a sample of 100 observations

**Error message**
```python
---------------------------------------------------------------------------
UnicodeDecodeError                        Traceback (most recent call last)
~/Desktop/Uni/Master/Master Thesis/Codes/Python Codes/main.py in 
----> 705 explainer = shap.TreeExplainer(model,background_data,model_output = 'probability')

/anaconda3/lib/python3.7/site-packages/shap/explainers/tree.py in __init__(self, model, data, model_output, feature_perturbation, **deprecated_options)
    119         self.feature_perturbation = feature_perturbation
    120         self.expected_value = None
--> 121         self.model = TreeEnsemble(model, self.data, self.data_missing, model_output)
    122         self.model_output = model_output
    123         #self.model_output = self.model.model_output # this allows the TreeEnsemble to translate model outputs types by how it loads the model

/anaconda3/lib/python3.7/site-packages/shap/explainers/tree.py in __init__(self, model, data, data_missing, model_output)
    724             self.original_model = model
    725             self.model_type = "xgboost"
--> 726             xgb_loader = XGBTreeModelLoader(self.original_model)
    727             self.trees = xgb_loader.get_trees(data=data, data_missing=data_missing)
    728             self.base_offset = xgb_loader.base_score

/anaconda3/lib/python3.7/site-packages/shap/explainers/tree.py in __init__(self, xgb_model)
   1324         self.read_arr('i', 29) # reserved
   1325         self.name_obj_len = self.read('Q')
-> 1326         self.name_obj = self.read_str(self.name_obj_len)
   1327         self.name_gbm_len = self.read('Q')
   1328         self.name_gbm = self.read_str(self.name_gbm_len)

/anaconda3/lib/python3.7/site-packages/shap/explainers/tree.py in read_str(self, size)
   1454 
   1455     def read_str(self, size):
-> 1456         val = self.buf[self.pos:self.pos+size].decode('utf-8')
   1457         self.pos += size
   1458         return val

UnicodeDecodeError: 'utf-8' codec can't decode byte 0xf4 in position 30: invalid continuation byte
```

**Attemped solution:**

add an exception in line *1455* in tree.py as follows

```python
def read_str(self, size):
        try:
            val = self.buf[self.pos:self.pos+size].decode('utf-8')
        except:
            val = self.buf[self.pos:self.pos+size].decode('latin-1')
        self.pos += size
        return val
```

--> problem then, is a conflict with byte size specified by self.name_gbm_len = 'Q' (*line 1325*)

**Comment by roye10** [comment]
Tried it, using the first example, as specified on the [homepage](https://github.com/slundberg/shap):

```python
X_example, y_example = shap.datasets.boston()
model = xgboost.train({'learning_rate': 0.01}, xgboost.DMatrix(X_example, label = y_example),100)
explainer_trying = shap.TreeExplainer(model)
```
and got the same error type:

```python
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xff in position 342: invalid start byte
```

**Comment by lrjball** [comment]
What version of Shap and xgboost are you using? A change to xgboost in the latest version broke this code in 0.35.0 but it should be fixed in 0.36.0, or by downgrading xgboost

**Comment by roye10** [comment]
Thanks! Yes, that did the trick. Sorry..will try to update all modules prior next time..
The shap version I was using was **0.35.0** and xgboost version **1.2.0**
